### DISCUSSION TITLE
Why can GCNConv not handle negative edge weights?

### DISCUSSION THREAD

**Discussion by dinhtrang24** [pyg-team/pytorch_geometric#2435] [Q&A] [answered]
Are there any reasons why graph convolutional operator GCNConv from the “Semi-supervised Classification with Graph Convolutional Networks” paper cannot handle negative edge weights? Is it because of mathematical complexity? Or are there any theoretical reasons/ assumptions, for example, if we violate them the model will fail? I have tried to find the answers in the paper, but it seems that the authors did not mention negative weight.

Thanks in advance!

**Comment by rusty1s** [comment, marked as answer]
The reason why `GCNConv` cannot support negative edge weights is due to the `pow(-0.5)` call on edge weights. If you do not perform such normalization, negative edge weights should work just fine.

**Reply by dinhtrang24** [reply]
Thanks a lot!
